en/3/01-externaldependencies.md

```diff
@@ -4,171 +4,236 @@ actions: ['checkAnswer', 'hints']
+            #[storage_mapper("zombie_owner")]
+            fn zombie_owner(&self, id: &usize) -> SingleValueMapper<ManagedAddress>;
+
+            #[storage_mapper("crypto_kitties_sc_address")]
+            fn crypto_kitties_sc_address(&self) -> SingleValueMapper<ManagedAddress>;
+
+            #[storage_mapper("owned_zombies")]
+            fn owned_zombies(&self, owner: &ManagedAddress) -> UnorderedSetMapper<usize>;
+        }
+      "lib.rs": |
+        #![no_std]
+
+        multiversx_sc::imports!();
+        multiversx_sc::derive_imports!();
+
+        mod storage;
+        mod zombie;
+        mod zombiefactory;
+        mod zombiefeeding;
+
+        #[multiversx_sc::contract]
+        pub trait ZombiesContract:
+            zombiefactory::ZombieFactory + zombiefeeding::ZombieFeeding + storage::Storage
+        {
+            #[init]
+            fn init(&self) {
+                self.dna_digits().set(16u8);
+            }
         }
+    answer: >
+      #![no_std]
+
+      multiversx_sc::imports!();
+      multiversx_sc::derive_imports!();
+
+      mod storage;
+      mod zombie;
+      mod zombiefactory;
+      mod zombiefeeding;
+
+      #[multiversx_sc::contract]
+      pub trait ZombiesContract:
+          zombiefactory::ZombieFactory + zombiefeeding::ZombieFeeding + storage::Storage
+      {
+          #[init]
+          fn init(&self) {
+              self.dna_digits().set(16u8);
+          }
 
+          #[endpoint]
+          fn set_crypto_kitties_sc_address(&self, address: ManagedAddress) {
+              self.crypto_kitties_sc_address().set(address);
+          }
       }
 ---
 
-Up until now, Solidity has looked quite similar to other languages like JavaScript.  But there are a number of ways that Ethereum DApps are actually quite different from normal applications.
-
-To start with, after you deploy a contract to Ethereum, it’s **_immutable_**, which means that it can never be modified or updated again.
+Up until now, Rust might have been a bit harder to deal with and the syntax hard to get, but fro this point we will experience what a strong programing language and a well designed framework can achieve when it comes to more complex elements for smart contract development and optimization. 
 
-The initial code you deploy to a contract is there to stay, permanently, on the blockchain. This is one reason security is such a huge concern in Solidity.  If there's a flaw in your contract code, there's no way for you to patch it later. You would have to tell your users to start using a different smart contract address that has the fix.
+To start with, after you deploy a contract to MultiversX, compared with other blockchains it is upgradable and hopefully one day soon manageable through versioning and even undeployable. Of course all these features come at a very complex syntax as you have seen by now, but as uncle be once told Peter: "With great power comes great responsability".
```

Comment:
I will formulate it something like: `Unlike on other blockchains, a contract deployed to MultiversX can be made upgradable and at some point, it will also be possible to undeploy it. Anthough contract can be made upgradable, it is not enforced. For some contracts it's even non-desirable.`
Just note about the undeplowment, but not do into debates and details about it